### PR TITLE
Minor: Add DataFrame test

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/8194

## Rationale for this change

Follow on to https://github.com/apache/arrow-datafusion/pull/8316 I noticed that one of the test was actually using the dataframe API, so it should also be put in the dataframe tests

## What changes are included in this PR?

Restore one tests into the dataframe tests

## Are these changes tested?
They are all tests

### CODE REVIEW

**Review by viirya — APPROVED** [review]

File: datafusion/core/tests/dataframe/mod.rs [hunk 1]
```diff
@@ -1597,3 +1597,28 @@ async fn test_array_agg() -> Result<()> {
...
+        .aggregate(vec![col("c1")], vec![sum(col("c2"))])?
+        .project(vec![col("c1"), sum(col("c2")).alias("total_salary")])?
+        .build()?;
+
+    let plan = state.optimize(&plan)?;
+    let physical_plan = state.create_physical_plan(&Arc::new(plan)).await?;
+    assert_eq!("c1", physical_plan.schema().field(0).name().as_str());
+    assert_eq!(
+        "total_salary",
+        physical_plan.schema().field(1).name().as_str()
```
Comment (viirya) [inline datafusion/core/tests/dataframe/mod.rs:1621]: Actually this test doesn't really use DataFrame API but creates logical plan and physical plan with `SessionState`. Maybe `physical-plan`? But there doesn't have `tests` module, so here is also good.

Comment (alamb) [reply datafusion/core/tests/dataframe/mod.rs:1621]: You are right -- I mistakenly assumed that the `scan_empty` was returning a DataFrame (it is actually returning a `LogicalPlanBuilder`): https://github.com/alamb/arrow-datafusion/blob/9a15bf89682728a277cc0a8f443fbd4ba8398ae6/datafusion/core/src/test_util/mod.rs#L133

I will see if I can find a better home for this

File: datafusion/core/src/physical_planner.rs [hunk 2]
```diff
@@ -2540,6 +2540,27 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn aggregate_with_alias() -> Result<()> {
```
Comment (alamb) [inline datafusion/core/src/physical_planner.rs:2544]: I found a better place for this test (with the other tests that are checking things about the physical plans)

File: datafusion/core/tests/dataframe/mod.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/tests/dataframe/mod.rs]: Moved in [2ea11a0](https://github.com/apache/arrow-datafusion/pull/8341/commits/2ea11a064b5ae7bb22b6a0e757382dee06e864e4)

File: datafusion/core/src/physical_planner.rs (hunk 2 above)
Comment (viirya) [reply datafusion/core/src/physical_planner.rs:2544]: Looks good. 👍

**Review by viirya — APPROVED** [review]

**Pull request merged**